src/util/records.py

```diff
@@ -1261,18 +1301,33 @@ def delete_unused(cr, *xmlids, **kwargs):
             ]
         )
         if sub:
-            cr.execute(
-                """
-                SELECT id
-                  FROM "{}" t
-                 WHERE id = ANY(%s)
-                   AND NOT EXISTS({})
-            """.format(table, sub),
-                [list(ids)],
+            query = format_query(
+                cr,
+                r"""
+                WITH _kids AS (
+                    {}
+                ),
+                _school AS ( -- grouped kids ¯\_(ツ)_/¯
+                    SELECT root as id, array_agg(child) as children
+                      FROM _kids
+                  GROUP BY root
+                )
+
+                SELECT t.id
+                  FROM {} t
+                  JOIN _school s
+                    ON s.id = t.id
+                 WHERE NOT EXISTS({})
+                """,
+                sql.SQL(kids_query),
+                table,
+                sql.SQL(sub),
```

Comment:
`kids_query` is already SQLStr, `sub` could be set explicitly here or before?
```suggestion
                kids_query,
                table,
                SQLStr(sub),
```